mrp_unbuild_origin_location/models/mrp_unbuild.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class MrpUnbuild(models.Model):
+    _inherit = "mrp.unbuild"
+
+    restore_rm_stock_in_origin_loc = fields.Boolean(
+        string="Restore Raw Materials to Consumed Location",
+        help="If enabled, the source location of the component stock move lines will be used"
+        " as the destination location for the unbuild move lines.",
+    )
+
+    def action_unbuild(self):
+        self.ensure_one()
+        if self.restore_rm_stock_in_origin_loc:
+            self = self.with_context(exact_location=True)
+        return super().action_unbuild()
+
+    def _prepare_move_line_vals(self, move, origin_move_line, taken_quantity):
+        vals = super()._prepare_move_line_vals(move, origin_move_line, taken_quantity)
+        if not self.env.context.get("exact_location"):
+            return vals
+        vals.update(
+            {
+                "location_id": origin_move_line.location_dest_id.id,
+                "location_dest_id": origin_move_line.location_id.id,
+            }
+        )
```

Comment:
Let's also take care of the owner.
```suggestion
        vals["owner_id"] = origin_move_line.owner_id.id
        if self.env.context.get("exact_location"):
            vals["location_id"] = origin_move_line.location_dest_id.id,
            vals["location_dest_id"] = origin_move_line.location_id.id,
```